src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/IObjectionService.java

```diff
@@ -10,13 +10,14 @@
 import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Attachment;
 import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Objection;
 import uk.gov.companieshouse.api.strikeoffobjections.model.patch.ObjectionPatch;
+import uk.gov.companieshouse.api.strikeoffobjections.model.response.ObjectionResponseDTO;
 import uk.gov.companieshouse.service.ServiceException;
 import uk.gov.companieshouse.service.ServiceResult;
 
 import javax.servlet.http.HttpServletResponse;
 
 public interface IObjectionService {
-    String createObjection(String requestId, String companyNumber, String ericUserId, String ericUserDetails);
+    ObjectionResponseDTO createObjection(String requestId, String companyNumber, String ericUserId, String ericUserDetails);
```

Comment:
I added a comment yesterday but it has disappeared, I don't think we should be returning the DTO as it is intended for converting to json for the api response. I'd prefer a new wrapper class that belongs in the same package as the service to wrap the results from createObjection.

Reply:
Not sure I follow - won't we just end up with more classes than we need? I'm not sure why we can't just use the DTO.

Reply:
Could we just return/use the 'Objection' entity class instead then (and map it). Similar to how the '_getObjection()_' method works?

Reply:
Then we just have more code than we need -do we really want extra mapping?

Reply:
The mapping is probably the best solution will look into this.

Reply:
We need to think of these components as being independent, by returning the DTO used in the controller layer, we are tying the service to the controller layer. If another team want to make changes to this code base in 2 years time, they will look at the createObjection return type and wonder why it is full of fields that aren't used and full of json annotations. Plus, more, smaller classes is better than fewer bigger classes, if each one has it's own purpose (separation of concerns I think it is called)

Reply:
We could just return Objection as martin said, it would be logical for a method called createObjection to return an objection.

Reply:
The mapper is already written so shouldn't be much more work.

Reply:
Service returns the objection and then in the controller this is mapped to the dto and returned.